### ISSUE TITLE
Mediapicker location filters don't work until full list is loaded for all pages.

### ISSUE THREAD

**Issue by PadineanuCosmin** [getgrav/grav-plugin-admin#1482]
Hi,

If a new mediapicker window opens it automatically opens with the filter "All Pages" so it loads all media available in all of the pages.
If the entire list hasn't loaded, user hasn't scrolled to the bottom several times to load the rest of the list until nothing is left, changing the filter to a specific location just reloads the same list as with the filter set to "All Pages".

If the user scrolls to the bottom several times until the entire list for filter "all Pages" is loaded and then chooses a page to filter by the filters work and restrict the results to the required page filter.

Can this be fixed to allow the filters to specific locations to work even thought there was no initial "full load" of the list with the "All Pages" filter? It makes working with lots of media(we have hundreds of images) very hard.

**Comment by rhukster** [comment]
Yes I've seen bugs like this. That field does need some JS love to fix some of these weird edge cases.

**Comment by dimitrilongo** [comment]
i had same issue makes `mediapicker` field useless (with lot of page and media in production). Any news ? Thanks
